app/controllers/purchases_controller.rb

```diff
@@ -28,22 +27,23 @@ def show
 
   private
 
-  def create_purchase(attributes, course, invoice_id)
-    Purchase.create!(attributes.merge(
+  def create_purchase_and_redirect(attributes, course, invoice_id)
+    purchase = Purchase.create!(attributes.merge(
       user: current_user,
-      course: @course,
+      course: course,
       invoice_id: invoice_id,
       paid: true,
       price: current_user.subscribed? ? course.discount_price : course.price,
       affiliate_tag: request.env['affiliate.tag'].presence
     ))
+    redirect_to purchase
   end
 
   def create_iugu_charge(iugu, course)
     response = iugu.charge.create(
       token: params[:token],
       customer_id: current_user.iugu_id,
-      months: params[:purchase][:installments],
+      months: params[:purchase][:installments].to_i,
```

Comment:
The API asks for an integer, but I didn't find any way to test whether the installments are really working. To be tested in production.